Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue

```diff
@@ -0,0 +1,457 @@
+const errorStore = useErrorStore();
+const userStore = useUserStore();
+
+const isVisible = ref(false);
+const isLoading = ref(false);
+const errorMessage = ref("");
+const errorType = ref<ActionType | null>(null);
+const accepted = ref(false);
+const dependent = ref<AddDependentRequest>({
+    firstName: "",
+    lastName: "",
+    dateOfBirth: "",
+    PHN: "",
+});
+
+const hasValidationErrors = computed(() => {
+    v$.value.accepted.$touch();
+    return v$.value.$invalid;
+});
+const isError = computed(
+    () => errorType.value !== null || errorMessage.value.length > 0
+);
+const isErrorDataMismatch = computed(
+    () => errorType.value === ActionType.DataMismatch
+);
+const isErrorNoHdid = computed(() => errorType.value === ActionType.NoHdId);
+const isErrorProtected = computed(
+    () => errorType.value === ActionType.Protected
+);
+const isDependentAlreadyAdded = computed(() =>
+    dependentStore.dependents.some(
+        (d) =>
+            d.dependentInformation.PHN ===
+            dependent.value.PHN.replace(/\s/g, "")
+    )
+);
+const minBirthdate = computed<IDateWrapper>(() =>
+    new DateWrapper().subtract(
+        Duration.fromObject({ years: configStore.webConfig.maxDependentAge })
+    )
+);
+const validations = computed(() => ({
+    dependent: {
+        firstName: {
+            required: helpers.withMessage(
+                "Given names are required.",
+                required
+            ),
+        },
+        lastName: {
+            required: helpers.withMessage("Last name is required.", required),
+        },
+        dateOfBirth: {
+            required: helpers.withMessage("Invalid date.", required),
+            minLength: helpers.withMessage("Invalid date.", minLength(10)),
+            minValue: helpers.withMessage(
+                `Dependent must be under the age of
+                        ${configStore.webConfig.maxDependentAge}.`,
+                (value: string) =>
+                    new DateWrapper(value).isAfter(minBirthdate.value)
+            ),
+            maxValue: helpers.withMessage("Invalid Date.", (value: string) =>
+                new DateWrapper(value).isBefore(new DateWrapper())
+            ),
+        },
+        PHN: {
+            required: helpers.withMessage("Valid PHN is required.", required),
+            minLength: helpers.withMessage(
+                "Valid PHN is required.",
+                minLength(12)
+            ),
+            validPersonalHealthNumber: helpers.withMessage(
+                "Personal Health Number must be valid.",
+                ValidationUtil.validatePhn
+            ),
+            isNew: helpers.withMessage(
+                "This dependent has already been added.",
+                () => !isDependentAlreadyAdded.value
+            ),
+        },
+    },
+    accepted: { isChecked: sameAs(true) },
+}));
+
+const v$ = useVuelidate(validations, { dependent, accepted });
+
+function addDependent(): void {
+    isLoading.value = true;
```

Comment:
This is a good component to use the loadingStore.

which works well with a single loading state. I haven't used it on the timeline components as we would be registering a dozen loaders which would trigger the loading event for a multitude of components.

this could be argued as a limitation of using enum type for loaders. however in Single loader components like page/form/non-repeating components it is a a good place to use the loadingStore.

https://github.com/bcgov/healthgateway/blob/features/15545/Apps/WebClient/src/NewClientApp/src/components/private/profile/ProfileView.vue has a good example of using the loadingStore.

In short you register to the loader through a computed property and then use the helper function to manage promises that should activate the loader, and bob's your uncle.

Reply:
Changed.